src/Webhook.php

```diff
@@ -0,0 +1,1044 @@
+     * @param string $event The event being raised
+     * @param CommonDBTM $item The item the event is being raised for
+     * @return void
+     */
+    public static function raise(string $event, CommonDBTM $item): void
+    {
+        global $DB;
+
+        // Ignore raising if the table doesn't exist (happens during install/update)
+        if (!$DB->tableExists(self::getTable())) {
+            return;
+        }
+
+        $supported = self::getAPIItemtypeData();
+        $supported_types = [];
+        foreach ($supported as $categories) {
+            foreach ($categories as $types) {
+                $supported_types = array_merge($supported_types, array_keys($types));
+            }
+        }
+
+        // Ignore raising if the item type is not supported
+        if (!in_array($item->getType(), $supported_types, true)) {
+            return;
+        }
+
+        $it = $DB->request([
+            'SELECT' => ['id'],
+            'FROM' => self::getTable(),
+            'WHERE' => [
+                'event' => $event,
+                'itemtype' => $item->getType(),
+                'is_active' => 1
+            ]
+        ]);
+        if ($it->count() === 0) {
+            return;
+        }
+
+        // Get data from the API once for all the webhooks
+        $webhook = new self();
+        $path = $webhook->getApiPath($item);
+
+        foreach ($it as $webhook_data) {
+            $webhook->getFromDB($webhook_data['id']);
+
+            $body = $webhook->getResultForPath($path, $event);
+            // Check if the item matches the webhook filters
+            if (!$webhook->itemMatchFilter($item)) {
+                continue;
+            }
+            $timestamp = time();
+            $headers = [
+                'X-GLPI-signature' => self::getSignature($body . $timestamp, $webhook->fields['secret']),
+                'X-GLPI-timestamp' => $timestamp
+            ];
+            $data = $webhook->fields;
+            $data['items_id'] = $item->getID();
+            $data['body'] = $body;
+            $data['headers'] = json_encode($headers);
+            self::send($data);
+        }
+    }
+
+    /**
+     * Send a webhook to the queue
+     * @param array $data The data for the webhook
+     * @return void
+     */
+    public static function send(array $data)
+    {
+        $queued_webhook = new QueuedWebhook();
+        $queued_webhooks_id = $queued_webhook->add([
+            'itemtype' => $data['itemtype'],
+            'items_id' => $data['items_id'],
+            'entities_id' => $data['entities_id'],
+            'webhooks_id' => $data['id'],
+            'url' => $data['url'],
+            'body' => $data['body'],
+            'event' => $data['event'],
+            'headers' => $data['headers'],
+        ]);
+        // Try sending the webhook immediately
+        if ($queued_webhooks_id !== false) {
+            QueuedWebhook::sendById($queued_webhooks_id);
+        }
```

Comment:
IMHO, this should not be done synchronously.

In the past, we had issues with some `QueuedNotification` that were sent synchronously and were sometine degrading performances when a ticket was created/updated, or were triggering an error that prevented, for instance, the mail collector to work properly.